Implement parsing/serialization for font-size-adjust

https://developer.mozilla.org/en-US/docs/Web/CSS/font-size-adjust

The grammar is `none | <number>`. This means that it can either be the keyword `none`, or a number.

[`values::specified::Number`](https://doc.servo.org/style/values/specified/struct.Number.html) will help with storing and parsing then number.

See https://github.com/servo/servo/wiki/Property-hacking-guide for instructions on how to work on such bugs.

File: https://github.com/servo/servo/blob/master/components/style/properties/longhand/font.mako.rs

@Manishearth, while running "./mach test-unit -p style", I bump into an error as follows:

error[E0308]: mismatched types
   --> /Users/jeremy/servo/components/style/lib.rs:155:5
    |
155 |     include!(concat!(env!("OUT_DIR"), "/properties.rs"));
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected mutable reference, found enum `core::result::Result`
    |
    = note: expected type `&mut cssparser::Parser<'_, '_>`
    = note:    found type `core::result::Result<_, _>`
    = note: this error originates in a macro outside of the current crate 

I'm wondering if I should do something more to pass the unit test? Seems that even setting products="none", the newly added property would still be tested.

--- a/components/style/properties/longhand/font.mako.rs
+++ b/components/style/properties/longhand/font.mako.rs
@@ -349,6 +349,55 @@ ${helpers.single_keyword("font-variant",
     }
 </%helpers:longhand>
 
+// https://www.w3.org/TR/css-fonts-3/#font-size-adjust-prop
+// FIXME: This prop should be animatable
+<%helpers:longhand products="none" name="font-size-adjust" animatable="False">
+    use values::NoViewportPercentage;
+    use values::computed::ComputedValueAsSpecified;
+    use values::specified::Number;
+
+    impl ComputedValueAsSpecified for SpecifiedValue {}
+    impl NoViewportPercentage for SpecifiedValue {}
+
+    #[derive(Clone, Debug, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum SpecifiedValue {
+        None,
+        Number(Number),
+    }
+
+    pub mod computed_value {
+        use style_traits::ToCss;
+        use std::fmt;
+
+        pub use super::SpecifiedValue as T;
+
+        impl ToCss for T {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                match *self {
+                    T::None => dest.write_str("none"),
+                    T::Number(number) => number.to_css(dest),
+                }
+            }
+        }
+    }
+
+    #[inline] pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::None
+    }
+
+    /// none | <number>
+    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+        use values::specified::Number;
+
+        if input.try(|input| input.expect_ident_matching("none")).is_ok() {
+            return Ok(SpecifiedValue::None);
+        }
+
+        Ok(SpecifiedValue::Number(try!(Number::parse_non_negative(input))))
+    }
+</%helpers:longhand>
+
 <%helpers:longhand products="gecko" name="font-synthesis" animatable="False">
     use std::fmt;
     use style_traits::ToCss;

